server/routerlicious/packages/services-utils/src/auth.ts

```diff
@@ -318,7 +318,13 @@ export function validateTokenScopeClaims(expectedScopes: string): RequestHandler
 			);
 		}
 
-		const claims = decode(token) as ITokenClaims;
+		let claims: ITokenClaims;
+		try {
+			claims = decode(token) as ITokenClaims;
+		} catch {
+			throw new NetworkError(401, "Invalid token");
```

Comment:
Correct. This is a middleware. If you throw here it will have to be caught by a global error catcher which has less functionality than directly responding with the network error.

Reply:
Also I don't understand why this is being changed in this PR. It has nothing to do with Socket connection

Reply:
I started with just category some errors of connect, and to distinguish eac. I noticed this 401 error somehow becomes 500 on server side, so added these in this PR: 
<img width="354" alt="image" src="https://github.com/microsoft/FluidFramework/assets/3719246/770ee4d8-22b1-4e13-a74d-deaca8904f82">


And when I read code in socket connect, it make me feel almost impossible to read/follow the code, so I decided to do the refactor, which lead to the majority change.

Given the auth code change is minimum, I decided to leave them there, so I don't have to create another PR